frontend/src/pages/settings/+Page.tsx

```diff
@@ -0,0 +1,221 @@
+  margin-bottom: 30px;
+`
+
+const Label = styled.label`
+  display: flex;
+  align-items: center;
+  margin: 10px;
+  width: 100%;
+`
+
+const Input = styled.input`
+  margin-left: 10px;
+  padding: 8px;
+  border: 1px solid #ccc;
+  border-radius: 4px;
+  flex-grow: 1;
+`
+
+const Button = styled.button`
+padding: 8px 16px;
+`
+
+const DistrictList = styled.ul`
+  padding: 0;
+  width: 100%;
+`
+
+const DistrictListItem = styled.li`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin: 10px 0;
+  padding: 10px;
+  background-color: #f8f9fa;
+  border-radius: 4px;
+`
+
+const StatusButtonGroup = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 10px;
+`
+
+const Message = styled.p`
+  color: grey;
+`
+
+function Page(): JSX.Element {
+  const authContext = useAuthContext()
+
+  const [settings, setSettings] = useState<MoodleSettings[]>([
+    { name: 'academic_year', value: '' },
+    { name: 'academic_year_short', value: '' },
+    { name: 'course_category', value: '' },
+    { name: 'base_course_id', value: '' }
+  ])
+  const [districts, setDistricts] = useState<SchoolDistrict[]>([])
+  const [newDistrictName, setNewDistrictName] = useState<string>('')
+  const [settingsMessage, setSettingsMessage] = useState<string | null>(null)
+  const [districtMessage, setDistrictMessage] = useState<string | null>(null)
+
+  const fetchSettings = async (): Promise<void> => {
+    try {
+      const moodleSettings = await ropeApi.getMoodleSettings()
+      const updatedSettings = settings.map(setting => {
+        const foundSetting = moodleSettings.find(s => s.name === setting.name)
+        return (foundSetting != null) ? { ...setting, value: foundSetting.value, id: foundSetting.id } : setting
+      })
+      setSettings(updatedSettings)
+    } catch (error) {
+      console.error('Error fetching districts', error)
+    }
+  }
+  const fetchDistricts = async (): Promise<void> => {
+    try {
+      const districtsFromApi = await ropeApi.getDistricts()
+      setDistricts(districtsFromApi)
+    } catch (error) {
+      console.error('Error fetching districts:', error)
+    }
+  }
+  useEffect(() => {
+    if (authContext.isAdmin) {
+      void fetchSettings()
+      void fetchDistricts()
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [authContext.isAdmin])
+
+  const handleInputChange = (index: number, value: string): void => {
+    const updatedSettings = settings.map((setting, i) =>
+      i === index ? { ...setting, value } : setting
+    )
+    setSettings(updatedSettings)
+  }
+
+  const saveSettings = async (): Promise<void> => {
+    try {
+      await Promise.all(settings.map(async setting => {
+        if (setting.id !== undefined) {
+          return await ropeApi.updateMoodleSettings(setting.id, { ...setting, value: setting.value })
+        } else {
+          return await ropeApi.createMoodleSetting({ name: setting.name, value: setting.value })
+        }
+      }))
```

Comment:
I think something is not quite right here. Currently, if you start from a fresh DB / page, save settings, and then try to update, you get an error because the page tries to `POST` vs `PUT`. Reloading the page allows you to do things / will cause it to `PUT` as expected, so whatever it is it has to do with that initial state :man_shrugging: .